profile.md

```diff
@@ -486,10 +486,13 @@ Examples values of the `scope` claim:
 
 ### Group Based Authorization: wlcg.groups 
 
-Authorization may be based on the `wlcg.groups` claim. The value of the `wlcg.groups` claim is an ordered JSON array of case-sensitive strings reflecting the VO groups of which the token subject is a member. 
+Authorization may be based on the `wlcg.groups` claim. The value of the `wlcg.groups` claim is an ordered JSON array of case-sensitive strings reflecting the assertion, by the issuer, of group membership of the subject of the token.
 
-`wlcg.groups `semantics are equivalent to existing VOMS groups. VOMS roles should be considered as optional (i.e. returned only if requested) `wlcg.groups`; selection of optional groups is discussed below.
+For authorization decisions, relying parties MUST NOT consider any group memberships that are **not** asserted by the OP in the token.  Particularly, membership in a child group (e.g. `/cms/uscms`) does not imply membership in the parent group (`/cms`).
 
+The token does not need to list all groups the user has access to; relying parties MUST utilize only the asserted groups in their given order.  The `wlcg.groups` claim is optional and may either be empty or not provided at all.  The user may provide input on the contents and ordering of this claim; this is covered in [Scope-based Group Selection](#Scope-based-Group-Selection).
```

Comment:
Note to writer: it is easier to review changes if the lines are not so long!

### Not all groups

There are a number of potential issues with this line.

     The token does not need to list all groups the user has access to

A minor issue with nomenclature: users do not "have access to" groups, users "are members of" groups.  Better would be:

     The token does not need to list all groups the user is a member of

or (avoiding the trailing proposition).

     The token does not need to list all groups of which the user is a member

Rephrasing, to follow RFC 2119

     The token MAY exclude groups of which the user is a member

### utilize only

In the phrase 

    relying parties MUST utilize only the asserted groups in their given order.

This is really two things: only asserted groups and those groups in the given order.  I would suggest it is split into two statements, to avoid confusion.  For example, does the word "only" apply narrowly (to "asserted groups") or broadly (to "asserted groups in their given order").

#### asserted groups

The term "MUST utilize only the asserted groups" is overly broad.  It could be construed to mean that no other information in the token may be used.  This is clearly nonsense, since explicit AuthZ statements (in the scope claim) are something that RPs should honour.

Perhaps this could be replace with something more direct; e.g.,

    relying parties MUST NOT infer any group-membership

#### given order

The phrase

    in their given order

is somewhat confusing, as there's no definition on what this order means.

The document already suggests that the list of groups is ordered, although this could be made clearer.

My recommendation would be to:
  1. Drop this phrase here: I don't think it adds anything.
  2. Create an issue, clarifying the ordering of groups; in particular, clearly stating that groups are provided in an order.  That issue could also indicate that the document should state what are the processing expectations (if any) on the RP for the order in which group are listed.

###  empty claim.

In the phrase:

    The `wlcg.groups` claim is optional and may either be empty or not provided at all.

There are a few potential issues with this phrase.

1. Claim values are valid JSON, and "empty" is not a valid JSON type.  I think an "empty array" is what is meant.

    the `wlcg.groups` claim value SHALL be a JSON array of JSON strings, with each item asserting membership of the group identified by that string.  As a special case, a token MAY omit the `wlcg.groups` claim when the value would be an empty array.

Reply:
Hi Paul,
I have tried to address your various comments and suggestions in this commit:

https://github.com/WLCG-AuthZ-WG/common-jwt-profile/pull/46/commits/4e689b2e12187b19aad5ef19b57ade2f88374caa

Replying to the separate items in this discussion:

* I agree we should have insisted on keeping line lengths practical and will try to do so for changes.
* `Not all groups` + `utilize only`: given that several relevant notions are already expressed better in the preceding paragraphs of the section, I have replaced the sentence with one that is inspired by your suggestion.
* `empty claim`: I have changed the text starting from your suggestion, but formulating it slightly differently.
* We probably need to have an issue opened for the document to `state what are the processing expectations (if any) on the RP for the order in which group are listed` as you suggested.

What do you think?